rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPoolPartition.java

```diff
@@ -58,40 +61,123 @@ public ObjectPoolPartition(ObjectPool<T> pool, PoolConfig config,
     }
   }
 
-  public BlockingQueue<Poolable<T>> getObjectQueue()
+  public void returnObject(Poolable<T> object)
   {
-    return objectQueue;
+    if (!objectFactory.validate(object.getObject())) {
+      log.debug(String.format("Invalid object for host %s removing %s ", object.getHost(), object));
+      decreaseObject(object);
+      // Compensate for the removed object. Needed to prevent endless wait when in parallel a borrowObject is called
+      increaseObjects(1, false);
+      return;
+    }
+
+    log.debug(String.format("Returning object %s to queue of host %s. Queue size: %d", object, object.getHost(), objectQueue.size()));
+    if (!objectQueue.offer(object)) {
+      log.error("Created more objects than configured: " + totalCount);
```

Comment:
nit: we can print queue size to for better debugging.

Reply:
Left that out as that is derivable from max-size config

Reply:
done